lib/linguist/languages.yml

```diff
@@ -1975,6 +1975,7 @@ JSX:
   type: programming
   group: JavaScript
   extensions:
+  - ".js"
```

Comment:
Currently, it's kind of impossible to do that research on Github because it ignores some important characters - but from my knowledge much more people use `.js` extension for writing JSX code than `.jsx`.

If you go to [React's official repo examples](https://github.com/facebook/react/wiki/Examples), you'll see that all of them uses JSX in `.js` files (and the entire ecosystem does - Redux etc). Based on that, I assume that if the creators of JSX use the `.js` extension, it's the primary extension for the language. But I can't ensure that.